angular-primeng-app/src/app/components/post-details/post-details.component.ts

```diff
@@ -31,41 +32,39 @@ import { SanitizerHtmlPipe } from "../../pipes/sanitizer-html.pipe";
     ToolbarModule,
     ButtonModule,
     InputSwitchModule,
-    SanitizerHtmlPipe
+    SanitizerHtmlPipe,
+    SearchDialogComponent
   ],
   templateUrl: './post-details.component.html',
   styleUrl: './post-details.component.scss'
 })
-export class PostDetailsComponent implements OnInit, OnDestroy {
+export class PostDetailsComponent implements OnInit {
   post$!: Observable<Post>;
   blogInfo!: BlogInfo;
+  blogId: string = "";
   blogName: string = "";
-	blogSocialLinks!: BlogLinks;
-	checked: boolean = true;
-	selectedTheme: string = "dark";
-  route: ActivatedRoute = inject(ActivatedRoute);
-  private blogService: BlogService = inject(BlogService);
+  checked: boolean = true;
+  selectedTheme: string = "dark";
   themeService: ThemeService = inject(ThemeService);
-  private querySubscription?: Subscription;
+  private blogService: BlogService = inject(BlogService);
 
-  @Input({ required: true }) slug!: string;
+  @Input({required: true})
+  set slug(slug: string) {
+    this.post$ = this.blogService.getSinglePost(slug);
+  }
 
   ngOnInit(): void {
-    this.querySubscription = this.blogService
-    .getBlogInfo()
-    .subscribe((data) => {
-      this.blogInfo = data;
-      this.blogName = this.blogInfo.title;
-    });
-    this.post$ = this.blogService.getSinglePost(this.slug);
+    this.blogService
+      .getBlogInfo()
+      .subscribe((data) => {
+        this.blogInfo = data;
+        this.blogId = this.blogInfo.id;
+        this.blogName = this.blogInfo.title;
+      });
```

Comment:
@czBalazs98 indentation should be corrected in general cases, in cases where it doesn't overpower your work it can be ok

Reply:
can you explain it more what do you think of here?

Reply:
@czBalazs98 it's not perfect here but it is ok because it's still possible top read and you have done more significant work so indentation is secondary. Just please keep it minimal in the future, best to be avoided all together.

Reply:
@czBalazs98 this is part of your work so it is ok